scss/components/_off-canvas.scss

```diff
@@ -6,176 +6,298 @@
-    transition: transform $offcanvas-transition-length $offcanvas-transition-timing;
-    backface-visibility: hidden;
-    z-index: 1;
-    padding-bottom: 0.1px; // Prevents margin collapsing, which would reveal the box shadow of the wrapper
-
-    @if has-value($maincontent-shadow) {
-      box-shadow: $maincontent-shadow;
-    }
+  // Hides overflow on body when an off-canvas panel is open.
+  .is-off-canvas-open {
+    overflow: hidden;
   }
 
-  // Click-to-exit overlay (generated by JavaScript)
-  .js-off-canvas-exit {
-    display: none;
+  // Off-canvas overlay (generated by JavaScript)
+  .js-off-canvas-overlay {
     position: absolute;
     top: 0;
     left: 0;
+
     width: 100%;
     height: 100%;
+
+    transition: opacity $offcanvas-transition-length $offcanvas-transition-timing, visibility $offcanvas-transition-length $offcanvas-transition-timing;
+
     background: $offcanvas-exit-background;
-    cursor: pointer;
-    transition: background $offcanvas-transition-length $offcanvas-transition-timing;
+
+    opacity: 0;
+    visibility: hidden;
+
+    overflow: hidden;
+
+    &.is-visible {
+      opacity: 1;
+      visibility: visible;
+    }
+
+    &.is-closable {
+      cursor: pointer;
+    }
   }
 }
 
-/// Adds basic styles for an off-canvas menu.
-@mixin off-canvas-base {
+// Adds basic styles for an off-canvas wrapper.
+@mixin off-canvas-wrapper() {
+  position: relative;
+  overflow: hidden;
+}
+
+/// Adds basic styles for an off-canvas panel.
+@mixin off-canvas-base(
+  $background: $offcanvas-background,
+  $transition: $offcanvas-transition-length $offcanvas-transition-timing,
+  $fixed: $offcanvas-fixed
+) {
   @include disable-mouse-outline;
-  position: absolute;
-  background: $offcanvas-background;
-  z-index: $offcanvas-zindex;
-  max-height: 100%;
-  overflow-y: auto;
-  transform: translateX(0) translateY(0);
+
+  @if $fixed == true {
+    position: fixed;
+  }
+  @else {
+    position: absolute;
+  }
+
+  z-index: $offcanvas-push-zindex;
+
+  transition: transform $transition;
+  backface-visibility: hidden;
+
+  background: $background;
+
+  // Overlap only styles.
+  &.is-transition-overlap {
+    z-index: $offcanvas-overlap-zindex;
+
+    &.is-open {
+      box-shadow: $offcanvas-shadow;
+    }
+  }
+
+  // Sets transform to 0 to show an off-canvas panel.
+  &.is-open {
+    transform: translate(0, 0);
+  }
 }
 
+/// Adds styles to position an off-canvas panel to the left/right/top/bottom.
 @mixin off-canvas-position(
   $position: left,
-  $size: $offcanvas-size,
-  $fixed: false,
-  $vertical-size: $offcanvas-vertical-size
+  $orientation: horizontal,
+  $size: if($orientation == horizontal, $offcanvas-size, $offcanvas-vertical-size)
 ) {
   @if $position == left {
-    left: -$size;
     top: 0;
+    left: 0;
+
     width: $size;
+    height: 100%;
+
+    transform: translateX(-$size);
+    overflow-y: auto;
+
+    // Sets the open position for the content
+    &.is-open ~ .#{$maincontent-class} {
```

Comment:
This is because we no longer apply classes to the optional wrapper, so we need a way to trigger the content transition too, and need to cater for people using multiple off canvas sections (eg http://brettmason.co.uk/offcanvas/double/double-canvas.html). Does that make sense?